procrastinate/utils.py

```diff
@@ -245,9 +245,14 @@ def utcnow() -> datetime.datetime:
 def parse_datetime(raw: str) -> datetime.datetime:
     try:
         # this parser is the stricter one, so we try it first
-        return dateutil.parser.isoparse(raw)
+        dt = dateutil.parser.isoparse(raw)
+        if not dt.tzinfo:
+            dt = dt.replace(tzinfo=datetime.timezone.utc)
+        return dt
     except ValueError:
         pass
     # this parser is quite forgiving, and will attempt to return
     # a value in most circumstances, so we use it as last option
-    return dateutil.parser.parse(raw)
+    dt = dateutil.parser.parse(raw)
+    dt = dt.replace(tzinfo=datetime.timezone.utc)
+    return dt
```

Comment:
Can `dateutil.parser.parse`` return something else than whatever `dateutil.parser.isoparse` returns in cases where ``dateutil.parser.isoparse`` does ?

Just trying to understand in what case this would behave differently if we removed the whole try/except block.

Reply:
(I'll merge but we can continue discussing it. Best case scenario, this will allow for a follow-up PR simplifying the code)

Reply:
Both functions should return a `datetime.datetime` object. However, the second one seem more unpredictable when given exotic formats, hence the first attempt with the stricter one. Removing the `isoparse` call break the test with the `+01:00` offset.

Reply:
Ok, perfect then. As long as we have the proof that removing the first call changes anything, I'm fine keeping both. Thanks for your thourough answer